logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetMessageAssetUseCase.kt

```diff
@@ -116,7 +118,15 @@ internal class GetMessageAssetUseCaseImpl(
                         ).fold({
                             kaliumLogger.e("There was an error downloading asset with id => ${assetMetadata.assetKey.obfuscateId()}")
                             // This should be called if there is an issue while downloading the asset
-                            updateAssetMessageDownloadStatus(Message.DownloadStatus.FAILED_DOWNLOAD, conversationId, messageId)
+                            if (it is NetworkFailure.ServerMiscommunication &&
+                                it.kaliumException is KaliumException.InvalidRequestError
+                                && it.kaliumException.isNotFound()
+                            ) {
+                                updateAssetMessageDownloadStatus(Message.DownloadStatus.NOT_FOUND, conversationId, messageId)
```

Comment:
question: how this new `NOT_FOUND` status aligns with the `Failure.isRetryNeeded`? Because that's also used to not retry the request in some cases, but for 404 `isRetryNeeded` is not set to `false`, only for `isInvalidRequestError` and `FederatedBackendFailure` - why?

Reply:
Not found is one of isInvalidRequestError so it's ok to handle it in that way. I want to differentiate error on UI